Frameworks/Foundation/NSRange.mm

```diff
@@ -49,33 +49,18 @@ static BOOL NSLocationInRange(NSUInteger idx, NSRange r) {
  @Status Interoperable
 */
 NSRange NSRangeFromString(NSString* s) {
-    static NSCharacterSet* trimChars = [NSCharacterSet characterSetWithCharactersInString:@"{}- "];
-
-    // Valid string begins with '{' and ends with '}'
-    if ([s characterAtIndex:0] == '{' && [s characterAtIndex:([s length] - 1)] == '}') {
-        NSArray* components = [s componentsSeparatedByString:@","];
-
-        // The two components are position and length - any other number of components is invalid.
-        if ([components count] != 2) {
-            return NSMakeRange(0, 0);
-        }
-
-        unsigned long long position;
-        NSScanner* scanner = [NSScanner scannerWithString:[components[0] stringByTrimmingCharactersInSet:trimChars]];
-        if (![scanner scanUnsignedLongLong:&position]) {
-            return NSMakeRange(0, 0);
-        }
-
-        unsigned long long length;
-        scanner = [NSScanner scannerWithString:[components[1] stringByTrimmingCharactersInSet:trimChars]];
-        if (![scanner scanUnsignedLongLong:&length]) {
-            return NSMakeRange(0, 0);
-        }
-
-        return NSMakeRange(position, length);
+    static NSCharacterSet* numbers = [NSCharacterSet characterSetWithCharactersInString:@"1234567890"];
+    NSScanner* scanner = [NSScanner scannerWithString:s];
+    unsigned long long position = 0;
+    unsigned long long length = 0;
+
+    [scanner scanUpToCharactersFromSet:numbers intoString:nullptr];
```

Comment:
Yes, it will just try and grab the first two numbers it finds and make a range out of them.

---
In reply to: [73620628](https://github.com/Microsoft/WinObjC/pull/820#discussion_r73620628) [](ancestors = 73620628)